google-api-client/src/main/java/com/google/api/client/googleapis/services/AbstractGoogleClientRequest.java

```diff
@@ -164,6 +165,16 @@ static class ApiClientVersion {
     }
 
     public String toString() {
+      // When running the application as a native image, append `-graalvm` to the
+      // version.
+      String imageCode = System.getProperty("org.graalvm.nativeimage.imagecode");
```

Comment:
Please use constants instead of string literals here.

Reply:
That's a valid point! However, this might come down to the question of whether it is worth having dependency on graal-sdk in this repo? If we use the latest version (22.1.0) then we'll see the following incompatibility: 
```
bad class file: /home/runner/.m2/repository/org/graalvm/sdk/graal-sdk/22.1.0/graal-sdk-22.1.0.jar(org/graalvm/nativeimage/ImageInfo.class)
Error:      class file has wrong version 55.0, should be 52.0
Error:      Please remove or make sure it appears in the correct subdirectory of the classpath.
Error:  -> [Help 1]
```
In which case, we may need to do something similar to https://github.com/googleapis/gax-java/pull/1671. This could still pose a slight risk since we still target Java 7. But open to hearing about what you think!

Reply:
I'm fine with either. @suztomo what do you think? Is it worth it to bring the whole setup for compiling with 11 and testing on 7 in this case?